internal/pkg/task/config_runner.go

```diff
@@ -104,3 +107,65 @@ func (r *ConfigRunner) validateDependencies() error {
 
 	return nil
 }
+
+// CheckNonZeroExitCode returns the status of the containers part of the given tasks.
+func (r *ConfigRunner) CheckNonZeroExitCode(tasks []*Task) error {
+	var essentialContainers []string
+	taskDefName := fmt.Sprintf("copilot-%s", r.GroupName)
+	taskDefinition, err := r.Starter.TaskDefinition(taskDefName)
+	if err != nil {
+		return fmt.Errorf("get task definition %s of a task", taskDefName)
+	}
+
+	for _, container := range taskDefinition.ContainerDefinitions {
+		if aws.BoolValue(container.Essential) {
+			essentialContainers = append(essentialContainers, aws.StringValue(container.Name))
+		}
+	}
+
+	taskARNs := make([]string, len(tasks))
+	for idx, task := range tasks {
+		taskARNs[idx] = task.TaskARN
+	}
+	describedTasks, describeErr := r.Starter.DescribeTasks(r.Cluster, taskARNs)
+	if describeErr != nil {
+		return describeErr
+	}
+	// Today we only use one container inside a stand-alone task, shall we keep the looping logic
+	// active for multiple containers or only add this logic when we have multiple containers inside a stand-alone task in the future.
+	for _, describedTask := range describedTasks {
+		for _, container := range describedTask.Containers {
+			if contains(essentialContainers, aws.StringValue(container.Name)) && aws.Int64Value(container.ExitCode) > 0 {
+				taskID, err := ecs.TaskID(aws.StringValue(describedTask.TaskArn))
+				if err != nil {
+					return err
+				}
+				return fmt.Errorf(HumanString(aws.StringValue(container.Name),
+					taskID,
+					aws.Int64Value(container.ExitCode)))
+			}
+		}
+	}
+	return nil
+}
+
+// contains checks if a string is present in a slice
+func contains(s []string, str string) bool {
```

Comment:
Can we use [`map[string]struct{}`](https://github.com/aws/copilot-cli/pull/3612/files#diff-887bb70e74d71d9ecbfe078d768a70bdd2f02043501ba27c8b254ac3a102e799R380) to avoid for loop?

Reply:
We have used `map[string]bool` for instead of `essentialContainers []string` so we do not need this function anymore.